lib/repositories/records/mongo_db.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'connections/mongo_db'
+require 'sport_type'
+
+module Repositories
+  module Records
+    class MongoDb
+      attr_accessor :years, :sport_types, :group_by
+
+      def distance_per_month(years:, sport_types:)
+        q = [{ '$match' => matcher(years:, sport_types:) }]
+        q << { '$group' => {
+          _id:      { year: '$year', month: '$month' },
+          distance: { '$sum' => '$distance' }
+        } }
+        q << { '$sort' => { distance: -1 } }
+        q << { '$limit' => LIMIT }
+
+        data = sessions.aggregate(q)
+        data.each_with_object({}) do |doc, h|
+          key = "#{doc['_id']['year']}-#{doc['_id']['month']}"
+          h[key] = doc['distance'] / 1000.0
+        end
+      end
+
+      def distance_per_week(years:, sport_types:)
+        q = [{ '$match' => matcher(years:, sport_types:) }]
+        q << { '$group' => {
+          _id:      { year: '$year', month: '$month', week: { '$isoWeek' => '$start_time' } },
+          distance: { '$sum' => '$distance' }
+        } }
+        q << { '$sort' => { distance: -1 } }
+        q << { '$limit' => LIMIT }
+
+        data = sessions.aggregate(q)
+        data.each_with_object({}) do |doc, h|
+          key = "#{doc['_id']['year']}-#{doc['_id']['month']}-#{doc['_id']['week']}"
+          h[key] = doc['distance'] / 1000.0
+        end
+      end
+
+      def distance(years:, sport_types:)
+        q = [{ '$match' => matcher(years:, sport_types:) }]
+        q << { '$sort' => { distance: -1 } }
+        q << { '$limit' => LIMIT }
+
+        data = sessions.aggregate(q)
+        data.each_with_object({}) do |doc, h|
+          key = start_time_key(doc)
+          h[key] = doc['distance'] / 1000.0
+        end
+      end
+
+      def average_pace(years:, sport_types:)
+        matcher_with_distance = matcher(years:, sport_types:).merge('distance' => { "$gt": 0 })
+
+        q = [{ '$match' => matcher_with_distance }]
+        q << { '$project' => {
+          _id: 0, id: 1, start_time: 1, duration: 1, timezone: 1, distance: 1
+        } }
+        q << { '$addFields' => {
+          average_pace: { '$divide' => ['$duration', '$distance'] }
```

Comment:
This produces the unit `ms/m` right?

Reply:
Yes basically, but this is equal to seconds per km.

Reply:
I :heart: calculations with units to convert them :-)
`ms/m * 1000/1000 = s/km`